connect-inject/handler.go

```diff
@@ -779,3 +639,8 @@ func findServiceAccountVolumeMount(pod *corev1.Pod) (corev1.VolumeMount, error)
 
 	return volumeMount, nil
 }
+
+func (h *Handler) InjectDecoder(d *admission.Decoder) error {
+	h.decoder = d
+	return nil
+}
```

Comment:
```suggestion
```

Feel free to let me know why this is needed, but I pulled the branch and (after fixing a bug in the endpoints_controller_test()) it looks like everything compiles and the tests run fine without it?

Maybe there's a discrepancy in libraries that I have vs yours?

btw I had to apply this patch to get it to build, I'm not sure what the state of the base branch is or what the plans are with merging it, but for the PR to pass tests I needed this:
```
kyle@kyles-mbp consul-k8s % git diff
diff --git a/connect-inject/endpoints_controller_test.go b/connect-inject/endpoints_controller_test.go
index 643ea0c..592e3ae 100644
--- a/connect-inject/endpoints_controller_test.go
+++ b/connect-inject/endpoints_controller_test.go
@@ -223,7 +223,7 @@ func TestProcessUpstreams(t *testing.T) {
                                DenyK8sNamespacesSet:  mapset.NewSetWith(),
                        }

-                       upstreams, err := ep.processUpstreams(tt.pod())
+                       upstreams, err := ep.processUpstreams(*tt.pod())
                        if tt.expErr != "" {
                                require.EqualError(t, err, tt.expErr)
                        } else {
```

Reply:
So this is a runtime dependency and is used by the Manager to provide a decoder to the handler to enable it to decode Kubernetes objects in the fly. While in the tests we can provision a decoder directly, this method is how Dependency Injection is done during runtime. Does this answer your question?

WRT to patch, yeah, the stacked PRs have caused this at the moment and a rebase should potentially fix it. Im holding off on the rebase till the endpoints controller gets merged, but it will be address in the code that does get merged.

Reply:
Cool, thanks for the clarification!